api/src/main/java/com/exelaration/abstractmemery/security/WebSecurity.java

```diff
@@ -0,0 +1,63 @@
+package com.exelaration.abstractmemery.security;
+
+import static com.exelaration.abstractmemery.constants.SecurityConstants.SIGN_UP_URL;
+
+import com.exelaration.abstractmemery.filters.JWTAuthenticationFilter;
+import com.exelaration.abstractmemery.filters.JWTAuthorizationFilter;
+import com.exelaration.abstractmemery.services.implementations.UserDetailsServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.CorsConfigurationSource;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+
+@EnableWebSecurity
+public class WebSecurity extends WebSecurityConfigurerAdapter {
+
+  @Autowired private UserDetailsServiceImpl userDetailsService;
+  private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+  public WebSecurity(
+      UserDetailsServiceImpl userDetailsService, BCryptPasswordEncoder bCryptPasswordEncoder) {
+    this.userDetailsService = userDetailsService;
+    this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+  }
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.cors()
+        .and()
+        .csrf()
+        .disable()
+        .authorizeRequests()
+        .antMatchers(HttpMethod.POST, SIGN_UP_URL, "/meme/", "/upload/")
```

Comment:
We shouldn't `permitAll` on `/meme` or `/upload/`.  Those should be available only to authenticated users.

Reply:
Can't guest users make memes as well?

Reply:
Let's say no.  Guests should only be able to browse whatever is public.